tests/rptest/tests/schema_registry_test.py

```diff
@@ -1346,6 +1359,16 @@ def bool_alpha(b: bool) -> str:
         self.logger.debug("Running client, expecting success")
         client.run()
 
+        batches_decompressed = self.redpanda.metric_sum(
+            metric_name=
+            "vectorized_kafka_schema_id_cache_batches_decompressed_total",
+            metrics_endpoint=MetricsEndpoint.METRICS)
+
+        if compression_type != TopicSpec.CompressionTypes.NONE and payload_class == "com.redpanda.CompressiblePayload":
+            assert batches_decompressed > 0
+        if compression_type == TopicSpec.CompressionTypes.NONE:
+            assert batches_decompressed == 0
+
```

Comment:
I do not know too much about the serde client used in this test, but I would expect the value from `vectorized_kafka_schema_id_cache_batches_decompressed_total` to be predictable enough to have something like:
```py
assert batches_decompressed == expected_non_zero_value
```

Reply:
The client isn't guaranteed to compress, although the 128 bytes of the same character should be convincing.